### ISSUE TITLE
java.lang.IllegalArgumentException: Cannot deserialize instance of `java.lang.Boolean` out of START_OBJECT token

### ISSUE THREAD

**Issue by idear1203** [Azure/autorest.java#599]
I meet the following error when adding `--generate-sync-async-clients`. But the generation can be successful if it doesn't contain `--generate-sync-async-clients`. If this is caused by Swagger spec, is it possible to provide more detailed error message, for example the line number?

The Swagger spec is: https://github.com/idear1203/azure-rest-api-specs-pr/tree/synapse_update_data_plane_specs/specification/synapse/data-plane/Microsoft.Synapse/preview/2020-02-01-preview

The generation configuration file: https://github.com/idear1203/azure-sdk-for-java-pr/blob/synapse_java_track_2_sdk/sdk/synapse/azure-analytics-synapse-accesscontrol/swagger/README.md

```
autorest --java --use:D:\code\AzureSDK\autorest.java --generate-client-as-impl --generate-sync-async-clients
AutoRest code generation utility [cli version: 3.0.6187; node: v12.13.1, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\Users\dongwwa\.autorest\@autorest_core@3.0.6282\node_modules\@autorest\core\dist' (3.0.6282)
   Loading local AutoRest extension '@autorest/java' (D:/code/AzureSDK/autorest.java)
   Loading local AutoRest extension '@autorest/java.vanilla' (D:\code\AzureSDK\autorest.java\javagen)
   Loading local AutoRest extension '@autorest/java.preprocessor' (D:\code\AzureSDK\autorest.java\javagen\..\preprocessor)
   Loading AutoRest extension '@autorest/modelerfour' (4.13.309->4.13.309)
java.lang.IllegalArgumentException: Cannot deserialize instance of `java.lang.Boolean` out of START_OBJECT token
 at [Source: UNKNOWN; line: -1, column: -1]
        at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3922)
        at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3873)
        at com.azure.autorest.extension.base.jsonrpc.Connection.lambda$process$3(Connection.java:285)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `java.lang.Boolean` out of START_OBJECT token
 at [Source: UNKNOWN; line: -1, column: -1]
        at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
        at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1442)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1216)        at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1126)        at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$BooleanDeserializer._parseBoolean(NumberDeserializers.java:262)
        at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$BooleanDeserializer.deserialize(NumberDeserializers.java:199)
        at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$BooleanDeserializer.deserialize(NumberDeserializers.java:175)
        at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3917)
        ... 7 more
```

**Comment by weidongxu-microsoft** [comment]
As workaround, please try `--generate-sync-async-clients=true`

**Comment by idear1203** [comment]
Thank you @weidongxu-microsoft ! This works!

```
autorest --java --use:D:\code\AzureSDK\autorest.java --generate-client-as-impl=true --generate-sync-async-clients=true
```

**Comment by weidongxu-microsoft** [comment]
Link to https://github.com/Azure/autorest.java/issues/601

**Comment by weidongxu-microsoft** [comment]
Should be fixed in 4.0.1 release
https://github.com/Azure/autorest.java/releases/tag/v4.0.1
